src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -255,60 +194,47 @@ public void clusterChanged(ClusterChangedEvent event) {
+                    danglingIndices.remove(danglingIndex);
+                }
+            }
+            try {
+                for (String indexName : nodeEnv.findAllIndices()) {
+                    if (newMetaData.hasIndex(indexName)) { // if we have the index on the metadata, its not dangled
+                        continue;
+                    }
+                    if (danglingIndices.contains(indexName)) { // already dangling, continue
+                        continue;
                     }
-                    // delete indices that are no longer part of the metadata
                     try {
-                        for (String indexName : nodeEnv.findAllIndices()) {
-                            // if we have the index on the metadata, don't delete it
-                            if (newMetaData.hasIndex(indexName)) {
-                                continue;
-                            }
-                            if (danglingIndices.containsKey(indexName)) {
-                                // already dangling, continue
-                                continue;
-                            }
-                            final IndexMetaData indexMetaData = loadIndexState(indexName);
-                            if (indexMetaData != null) {
-                                if(autoImportDangled.shouldImport()){
-                                    logger.info("[{}] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [{}]", indexName, autoImportDangled);
-                                    danglingIndices.put(indexName, new DanglingIndex(indexName, null));
-                                } else if (danglingTimeout.millis() == 0) {
-                                    logger.info("[{}] dangling index, exists on local file system, but not in cluster metadata, timeout set to 0, deleting now", indexName);
-                                    indicesService.deleteIndexStore("dangling index with timeout set to 0", indexMetaData, state);
-                                } else {
-                                    logger.info("[{}] dangling index, exists on local file system, but not in cluster metadata, scheduling to delete in [{}], auto import to cluster state [{}]", indexName, danglingTimeout, autoImportDangled);
-                                    danglingIndices.put(indexName,
-                                            new DanglingIndex(indexName,
-                                                    threadPool.schedule(danglingTimeout,
-                                                            ThreadPool.Names.SAME,
-                                                            new RemoveDanglingIndex(indexMetaData))));
-                                }
-                            }
+                        final IndexMetaData indexMetaData = loadIndexState(indexName);
+                        if (indexMetaData != null) {
```

Comment:
we can debug log it I guess, might happen is someone is copying things over for example, and the state is not yet copied